controllers/masteruserrecord/masteruserrecord_controller_test.go

```diff
@@ -234,36 +255,216 @@ func TestCreateMultipleUserAccountsSuccessful(t *testing.T) {
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			HasConditions().
+			HasStatusUserAccounts(commontest.MemberClusterName).
+			HasFinalizer()
+		AssertThatCountersAndMetrics(t).
+			HaveUsersPerActivationsAndDomain(toolchainv1alpha1.Metric{
+				"1,internal": 1, // unchanged
+			}).
+			HaveMasterUserRecordsPerDomain(toolchainv1alpha1.Metric{
+				string(metrics.Internal): 1, // unchanged
+			})
+	})
+
+	t.Run("mur is being moved - new account should be created", func(t *testing.T) {
+		murCopy := mur.DeepCopy()
+		murCopy.Status.UserAccounts = []toolchainv1alpha1.UserAccountStatusEmbedded{
+			{
+				Cluster: toolchainv1alpha1.Cluster{
+					Name: commontest.Member2ClusterName,
+				},
+			},
+		}
+		memberClient := commontest.NewFakeClient(t)
+		memberClient2 := commontest.NewFakeClient(t, newUserAccount(namespacedName(commontest.MemberOperatorNs, mur.Name), mur))
+		hostClient := commontest.NewFakeClient(t, murCopy, spaceBinding, space, sharedSpaceBinding, sharedSpace, toolchainStatus)
+		InitializeCounters(t, toolchainStatus)
+
+		cntrl := newController(hostClient, s, ClusterClient(commontest.MemberClusterName, memberClient), ClusterClient(commontest.Member2ClusterName, memberClient2))
+
+		// when
+		result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+		// then
+		require.NoError(t, err)
+		assert.Empty(t, result.RequeueAfter)
+		uatest.AssertThatUserAccount(t, "john", memberClient).
+			Exists().
+			MatchMasterUserRecord(mur).
+			HasLabelWithValue(toolchainv1alpha1.TierLabelKey, "deactivate30")
+		uatest.AssertThatUserAccount(t, "john", memberClient2).
+			Exists().
+			MatchMasterUserRecord(mur).
+			HasLabelWithValue(toolchainv1alpha1.TierLabelKey, "deactivate30")
+		murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+			HasConditions(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")).
+			HasStatusUserAccounts(commontest.Member2ClusterName).
+			HasFinalizer()
+		AssertThatCountersAndMetrics(t).
+			HaveUsersPerActivationsAndDomain(toolchainv1alpha1.Metric{
+				"1,internal": 1, // unchanged
+			}).
+			HaveMasterUserRecordsPerDomain(toolchainv1alpha1.Metric{
+				string(metrics.Internal): 1, // unchanged
+			})
+
+		t.Run("the previous account should be deleted", func(t *testing.T) {
+			// when
+			result, err := cntrl.Reconcile(context.TODO(), newMurRequest(mur))
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, 10*time.Second, result.RequeueAfter)
+			uatest.AssertThatUserAccount(t, "john", memberClient).
+				Exists().
+				MatchMasterUserRecord(mur).
+				HasLabelWithValue(toolchainv1alpha1.TierLabelKey, "deactivate30")
+			uatest.AssertThatUserAccount(t, "john", memberClient2).
+				DoesNotExist()
+			murtest.AssertThatMasterUserRecord(t, "john", hostClient).
+				HasConditions(toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "")).
+				HasStatusUserAccounts(commontest.Member2ClusterName, commontest.MemberClusterName).
+				HasFinalizer()
```

Comment:
> maybe I'm making some confusion here, but isn't the parent test moving the useraccount from memberClient to memberClient2 ? why should the useraccount be gone from memberClient2 ?

it's the other way around - we are moving from `memberClient2` to `memberClient`

> Also why the MUR status has both member clusters : HasStatusUserAccounts(commontest.Member2ClusterName, commontest.MemberClusterName). Shouldn't it be only HasStatusUserAccounts(commontest.Member2ClusterName) ?

this is the first step when a new account is created. Deletion happens in the next reconcile

Reply:
I improved the names of the sub-tests in [b5c27cc](https://github.com/codeready-toolchain/host-operator/pull/872/commits/b5c27cc0a44b189da4d6a5f7ea8f29b60cfb2b68)

Reply:
Thanks for clarifying that and renaming the test cases 

I guess I got a bit confused because:
- the original mur was created at the beginning of the test and has `member-cluster` when gets here:https://github.com/codeready-toolchain/host-operator/pull/872/files#diff-3759f9ea272110e6b2908772b8be4bf656b100aa84cf48ad0eea8bd65cc8aadfR365
- it passes `murCopy` only to the hostClient https://github.com/codeready-toolchain/host-operator/pull/872/files#diff-3759f9ea272110e6b2908772b8be4bf656b100aa84cf48ad0eea8bd65cc8aadfR375
- while uses `mur` in `memberClient2` https://github.com/codeready-toolchain/host-operator/pull/872/files#diff-3759f9ea272110e6b2908772b8be4bf656b100aa84cf48ad0eea8bd65cc8aadfR374 and the reconcile uses `mur` as well https://github.com/codeready-toolchain/host-operator/pull/872/files#diff-3759f9ea272110e6b2908772b8be4bf656b100aa84cf48ad0eea8bd65cc8aadfR381

Reply:
I see and I fully agree that it's confusing. I changed the test so it moves from `member-cluster` to `member2-cluster` (which is a bit more intuitive) and also improved the variable assignment and usage. 8c7265f6

Reply:
Nice! Much better now IMO! Thanks for improving this test!